tag_export.py

```diff
@@ -71,24 +71,36 @@ def __init__(self, timdex_dataset: TIMDEXDataset):
             self.timdex_dataset = timdex_dataset
 
         def parse_tags_from_record(
-            self, record: dict, tags: list[str]
+            self,
+            record: dict,
+            tags: list[str],
+            tag_value_regex: str,
         ) -> Iterator[tuple[str, str, str]]:
             """Load MARC XML and yield requested tags."""
             record_xml = etree.fromstring(record["source_record"])
             record_marc = marcalyx.Record(record_xml)
 
             for tag in tags:
-                for field in record_marc.field(tag):
+                for field_value in record_marc.field(tag):
+                    field_value_str = str(field_value)
+
+                    # apply regex to tag values if provided
+                    if tag_value_regex != "" and not re.match(
+                        tag_value_regex, field_value_str
+                    ):
+                        continue
```

Comment:
Most of the changes are leading to this: applying regex to the tag values before we yield them.